tests/reclaim_policy_test.go

```diff
@@ -0,0 +1,274 @@
+
+			By("verifying pod count as 0")
+			err = Client.waitForPods(applicationNamespace, label, corev1.PodRunning, 0)
+			Expect(err).To(BeNil(), "while verifying pod count")
+		})
+	})
+
+	When("pvc with storageclass reclaim-openebs-rwx is deleted ", func() {
+		It("should delete the pvc but not NFS service related artifacts", func() {
+			var err error
+
+			// Store bounded PVC object
+			claimedPVCObj, err = Client.getPVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while fetching pvc {%s} in namespace {%s}", pvcName, applicationNamespace)
+
+			By("deleting pvc")
+			err = Client.deletePVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while deleting pvc {%s} in namespace {%s}", pvcName, applicationNamespace)
+
+			isPVCDeleted := false
+			for retries := 0; retries < maxRetryCount; retries++ {
+				_, err := Client.getPVC(applicationNamespace, pvcName)
+				if err != nil && k8serrors.IsNotFound(err) {
+					isPVCDeleted = true
+					break
+				}
+				time.Sleep(time.Second * 5)
+			}
+			Expect(isPVCDeleted).To(BeTrue(), "pvc should be deleted from cluster")
+
+			pvObj, err := Client.getPV(claimedPVCObj.Spec.VolumeName)
+			Expect(err).To(BeNil(), "while fetching pv {%s} details", claimedPVCObj.Spec.VolumeName)
+			Expect(pvObj.DeletionTimestamp).To(BeNil(), "deletion timestamp shouldn't be set on pv {%s}", claimedPVCObj.Spec.VolumeName)
+
+			backendNFSName := "nfs-" + claimedPVCObj.Spec.VolumeName
+			deploymentObj, err := Client.getDeployment(openebsNamespace, backendNFSName)
+			Expect(err).To(BeNil(), "while fetching deployment {%s} in namespace {%s}", backendNFSName, openebsNamespace)
+			Expect(deploymentObj.DeletionTimestamp).To(BeNil(), "deletion timestamp shouldn't be set on deployment {%s}", backendNFSName)
+
+			svcObj, err := Client.getService(openebsNamespace, backendNFSName)
+			Expect(err).To(BeNil(), "while fetching service {%s} in namespace {%s}", backendNFSName, openebsNamespace)
+			Expect(svcObj.DeletionTimestamp).To(BeNil(), "deletion timestamp shouldn't be set on service {%s}", backendNFSName)
+
+			backendPVCObj, err := Client.getPVC(openebsNamespace, backendNFSName)
+			Expect(err).To(BeNil(), "while fetching backend pvc {%s} in namespace {%s}", backendNFSName, openebsNamespace)
+			Expect(backendPVCObj.DeletionTimestamp).To(BeNil(), "deletion timestamp shouldn't be set on backend pvc {%s}", backendNFSName)
+			Expect(backendPVCObj.Status.Phase).To(Equal(corev1.ClaimBound), "while verifying backed PVC claim phase")
+
+			backendPVObj, err := Client.getPV(backendPVCObj.Spec.VolumeName)
+			Expect(err).To(BeNil(), "while fetching pv {%s}", backendPVCObj.Spec.VolumeName)
+			Expect(backendPVObj.DeletionTimestamp).To(BeNil(), "deletion timestamp shouldn't be set on backend pv {%s}", backendPVObj.Name)
+			Expect(backendPVObj.Status.Phase).To(Equal(corev1.VolumeBound), "while verifying backed PV bound phase")
+
+		})
+	})
+
+	When("deleting NFS service related resources", func() {
```

Comment:
can we change it to something like, `deleting NFS server resources` or `cleaning up nfs volume`?

Reply:
updated